src/server/controlchan/commands/pasv.rs

```diff
@@ -52,14 +51,21 @@ impl Pasv {
 
         let mut listener: io::Result<TcpListener> = Err(io::Error::new(io::ErrorKind::InvalidInput, "Bind retries cannot be 0"));
 
-        let mut rng = OS_RNG.lock().await;
+        let lock = OS_RNG.lock().await;
         for _ in 1..BIND_RETRIES {
-            let port = rng.next_u32() % rng_length as u32 + passive_ports.start as u32;
+            let random_u32 = {
+                let mut data = [0; 4];
+                getrandom::getrandom(&mut data).expect("Error generating random port");
```

Comment:
I wouldn't want us to panic with `.expect`, not sure in which cases an error could happen here?

Reply:
[The previous implementation would also panic](https://docs.rs/rand_core/0.6.1/src/rand_core/os.rs.html#61-70), and [it's probably ok to do so](https://docs.rs/getrandom/0.2.3/getrandom/#error-handling)

Reply:
OK Fair enough